From 33849c250f9b4800ea40c506a88d9c184d6b5e46 Mon Sep 17 00:00:00 2001
From: ksrt12 <kazakov12stepan2012@live.ru>
Date: Wed, 18 Nov 2015 11:13:25 +0600
Subject: [PATCH 4/5] add cleanspec

---
 CleanSpec.mk | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/CleanSpec.mk b/CleanSpec.mk
index e9d4455..988d721 100644
--- a/CleanSpec.mk
+++ b/CleanSpec.mk
@@ -317,6 +317,12 @@ $(call add-clean-step, rm -rf $(PRODUCT_OUT)/obj/APPS/*)
 # Added new build props
 $(call add-clean-step, rm -rf $(PRODUCT_OUT)/system/build.prop)
 
+# Kernel
+$(call add-clean-step, rm -rf $(PRODUCT_OUT)/obj/KERNEL_OBJ/arch/arm/boot/*)
+$(call add-clean-step, rm -rf $(PRODUCT_OUT)/obj/KERNEL_OBJ/vmlinux)
+$(call add-clean-step, rm -rf $(PRODUCT_OUT)/obj/KERNEL_OBJ/vmlinux.o)
+$(call add-clean-step, rm -rf $(PRODUCT_OUT)/system/lib/modules/*)
+
 # ************************************************
 # NEWER CLEAN STEPS MUST BE AT THE END OF THE LIST
 # ************************************************
-- 
2.5.0

