From 86634f26a972c479e464f0e11e9a816a7a3b6675 Mon Sep 17 00:00:00 2001
From: ksrt12 <kazakov12stepan2012@live.ru>
Date: Wed, 18 Nov 2015 16:51:12 +0600
Subject: [PATCH 5/5] Add chromium prebuilt

---
 envsetup.sh | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/envsetup.sh b/envsetup.sh
index d80e95c..dbc4337 100644
--- a/envsetup.sh
+++ b/envsetup.sh
@@ -547,6 +547,13 @@ function lunch()
 
     echo
 
+    if [[ $USE_PREBUILT_CHROMIUM -eq 1 ]]; then
+        chromium_prebuilt
+    else
+        # Unset flag in case user opts out later on
+        export PRODUCT_PREBUILT_WEBVIEWCHROMIUM=""
+    fi
+
     set_stuff_for_environment
     printconfig
 }
@@ -1720,6 +1727,19 @@ function make()
     return $ret
 }
 
+function chromium_prebuilt() {
+    T=$(gettop)
+    export TARGET_DEVICE=$(get_build_var TARGET_DEVICE)
+    hash=$T/prebuilts/chromium/$TARGET_DEVICE/hash.txt
+ 
+    if [ -r $hash ] && [ $(git --git-dir=$T/external/chromium_org/.git --work-tree=$T/external/chromium_org rev-parse --verify HEAD) == $(cat $hash) ]; then
+        export PRODUCT_PREBUILT_WEBVIEWCHROMIUM=yes
+        echo "** Prebuilt Chromium is up-to-date; Will be used for build **"
+    else
+        export PRODUCT_PREBUILT_WEBVIEWCHROMIUM=no
+        echo "** Prebuilt Chromium out-of-date/not found; Will build from source **"
+    fi
+}
 
 
 if [ "x$SHELL" != "x/bin/bash" ]; then
-- 
2.5.0

